src/features/harmful-apis.js

```diff
@@ -0,0 +1,450 @@
+import ContentFeature from '../content-feature'
+import { DDGReflect, stripVersion } from '../utils'
+import { hasMozProxies, wrapMethod, wrapProperty } from '../wrapper-utils'
+
+/**
+ * block some Permission API queries. The set of available permissions is quickly changing over time. Up-to-date lists can be found here:
+ * - Chromium: https://chromium.googlesource.com/chromium/src/+/refs/heads/main/third_party/blink/renderer/modules/permissions/permission_descriptor.idl
+ * - Gecko: https://searchfox.org/mozilla-central/source/dom/webidl/Permissions.webidl#10
+ * - WebKit: https://github.com/WebKit/WebKit/blob/main/Source/WebCore/Modules/permissions/PermissionName.idl
+ * @param {string[]} permissions permission names to auto-deny
+ */
+function filterPermissionQuery (permissions) {
+    if (!permissions || permissions.length === 0) {
+        return
+    }
+    wrapMethod(globalThis.Permissions.prototype, 'query', async function (nativeImpl, queryObject) {
+        // call the original function first in case it throws an error
+        const origResult = await DDGReflect.apply(nativeImpl, this, [queryObject])
+
+        if (permissions.includes(queryObject.name)) {
+            return {
+                name: queryObject.name,
+                state: 'denied',
+                status: 'denied'
+            }
+        }
+        return origResult
+    })
+}
+
+/**
+ * Blocks some privacy harmful APIs.
+ */
+export default class HarmfulApis extends ContentFeature {
+    init () {
+        // @ts-expect-error linting is not yet seet up for worker context
+        /** @type Navigator | WorkerNavigator */
+        this.navigatorPrototype = globalThis.Navigator?.prototype || globalThis.WorkerNavigator?.prototype
+
+        this.removeDeviceOrientationEvents(this.getFeatureSetting('deviceOrientation'))
+        this.blockGenericSensorApi(this.getFeatureSetting('GenericSensor'))
+        this.filterUAClientHints(this.getFeatureSetting('UaClientHints'))
+        this.removeNetworkInformationApi(this.getFeatureSetting('NetworkInformation'))
+        this.blockGetInstalledRelatedApps(this.getFeatureSetting('getInstalledRelatedApps'))
+        this.removeFileSystemAccessApi(this.getFeatureSetting('FileSystemAccess'))
+        this.blockWindowPlacementApi(this.getFeatureSetting('WindowPlacement'))
+        this.blockWebBluetoothApi(this.getFeatureSetting('WebBluetooth'))
+        this.blockWebUsbApi(this.getFeatureSetting('WebUsb'))
+        this.blockWebSerialApi(this.getFeatureSetting('WebSerial'))
+        this.blockWebHidApi(this.getFeatureSetting('WebHid'))
+        this.blockWebMidiApi(this.getFeatureSetting('WebMidi'))
+        this.removeIdleDetectionApi(this.getFeatureSetting('IdleDetection'))
+        this.removeWebNfcApi(this.getFeatureSetting('WebNfc'))
+        this.filterStorageManagerApi(this.getFeatureSetting('StorageManager'))
+    }
```

Comment:
`this.getFeatureSetting()` can return undefined - but all of your methods expect to be able to check 'settings.protected'

Is this because you're going to always supply defaults in the config? Even if so, until we have schema verification + runtime checking I'd be tempted to convert everything to 

```js
if (!settings?.protected) {

}
```

Feel free to ignore if there's a reason this wouldn't be needed

Reply:
Right, the [default config](https://github.com/duckduckgo/privacy-configuration/blob/harmful-apis/features/harmful-apis.json) is supposed to always have these keys. If I understand correctly, this will always be loaded, right?
I think your suggestion makes sense though, I'll change it.